Fix indexedDB run dependency

lionel- opened the pull request:
`removeRunDependency()` should only be run after `addRunDependency()`. This is checked by an assertion in debug builds.

`Fetch.staticInit()` currently runs `addRunDependency()` _after_ calling `openDatabase()`. The method creates an XHR request with `removeRunDependency()` called asynchronously in onsuccess and onerror fallbacks. This works most of the time except when `onerror()` is called synchronously via try/catch when the request can't be created.

The first commit fixes this by creating the run dependency first. The second commit simplifies control flow a bit since the conditioning on `isMainThread` is no longer necessary.

lionel- committed:
Consistently run `addRunDependency()` in `Fetch.staticInit()`

lionel- committed:
Remove dangling condition

georgestagg commented on src/Fetch.js:
```diff
@@ -45,35 +46,26 @@ var Fetch = {
 #endif // ~FETCH_SUPPORT_INDEXEDDB
-
-#if FETCH_SUPPORT_INDEXEDDB
-    if (typeof ENVIRONMENT_IS_FETCH_WORKER == 'undefined' || !ENVIRONMENT_IS_FETCH_WORKER) addRunDependency('library_fetch_init');
```
I am a little confused about how `isMainThread` and `ENVIRONMENT_IS_FETCH_WORKER` were supposed to interact in the original version and a little worried since they have been removed entirely in the fix.

At first glance I thought the intention was that when `ENVIRONMENT_IS_FETCH_WORKER` is set, then `add`/`removeRunDependency` should not be run at any point, but then that cannot be right since I can also see that `isMainThread` is just set to `true` all the time?

Do we know what it means for `ENVIRONMENT_IS_FETCH_WORKER` to be set? Before continuing we should be sure that it is OK to run `add`/`removeRunDependency` even when `ENVIRONMENT_IS_FETCH_WORKER` is true.

georgestagg replied:
Okay, so I did some more searching. It looks like the branching used to be required when an option for fastcomp existed named `USE_FETCH_WORKER`. The option was removed in 5bbb777c17801c2a965124afcf069957d3eafd2c, the fetch worker environment is no more, and as far as I can tell `ENVIRONMENT_IS_FETCH_WORKER` is always undefined.

So, we're safe to remove the branching code and simply always run the `add`/`removeRunDependency` as you have implemented it.

In addition, I think removing this means there are no other instances of `ENVIRONMENT_IS_FETCH_WORKER`, so I think we can also remove the definition in [`src/closure-externs/closure-externs.js`](https://github.com/emscripten-core/emscripten/blob/main/src/closure-externs/closure-externs.js#L220).

lionel- replied:
I think the rundep model requires each "add" to be paired with a "remove" in the control flow. So the branching into remove paths must match the branching into add paths. This was the case in earlier commits of the history and then the symmetry got lost, which was either a bug or an obfuscation.

So I felt confident to remove the conditioning on `ENVIRONMENT_IS_FETCH_WORKER` without diving too much into the history or reasons for the current state of the code.

georgestagg replied:
I was just confused enough by it that I wanted to know more! OK, let's stick with your patch as-is and not worry about other use of `ENVIRONMENT_IS_FETCH_WORKER` or `closure-externs.js`. The maintainers can always clean it up if they wish to do so.

lionel- replied:
yup I think I should have proactively explained my reasoning here in a pre-review.

georgestagg reviewed (commented):
The fix makes sense to me.

I think it should be OK as long as we are sure that one of either `onsuccess()` or `onerror()` is always run so that one of the two always removes the run dependency, which does seem to be the case. I also have a question about `ENVIRONMENT_IS_FETCH_WORKER` which I have written below.

Were you able to get the built in Emscripten unit tests running?

lionel- committed:
Remove unused `ENVIRONMENT_IS_FETCH_WORKER` declaration

lionel- commented:
I've now added a unit test that compiles an Emscripten runtime that includes `Fetch` and then runs it under Node to check that initialisation works.

I tried running the test suite but it takes a long time and there are many failures. So it's not really practical to see if I broke something. However I ran the original tests that were added when Fetch was implemented and they run fine: `test/runner "browser.test_fetch*"`.

I've also removed the dangling `ENVIRONMENT_IS_FETCH_WORKER` declaration and checked the only closure-externs test I could find: `test/runner other.test_closure_externs`.
